src/wallet/scriptpubkeyman.cpp

```diff
@@ -2852,6 +2852,11 @@ bool DescriptorScriptPubKeyMan::CanUpdateToWalletDescriptor(const WalletDescript
         return false;
     }
 
+    if (!descriptor.descriptor->IsRange()) {
+        // Skip range check for non-range descriptors
+        return true;
+    }
+
     if (descriptor.range_start > m_wallet_descriptor.range_start ||
```

Comment:
Nit: Tying these 2 if blocks with an `else if` would be nice, although not necessary because the first if returns.

Reply:
I wasn't satisfied with how this would look, so I left it as-is.